### ISSUE TITLE
accuracy issue (using the 20 newsgroups example model)

### ISSUE THREAD

**Issue by mikkokotila** [keras-team/keras#10624]
The post in Keras blog [here](https://blog.keras.io/using-pre-trained-word-embeddings-in-a-keras-model.html) yields a 95%  validation accuracy after 2 epochs. Given the example code, I can't reproduce this (in fact the validation accuracy will not go over 75% and loss is very high as well). 

I'm following the example code here:

https://blog.keras.io/using-pre-trained-word-embeddings-in-a-keras-model.html

...with the exception that I open the GloVe vector file with encoding='utf-8' as it fails otherwise. The link is provided in the example blog post itself.

**Comment by mikkokotila** [comment]
Tried with a different machine and the results are same: 

```
Indexing word vectors.
Found 400000 word vectors.
Processing text dataset
Found 19997 texts.
Found 174074 unique tokens.
Shape of data tensor: (19997, 1000)
Shape of label tensor: (19997, 20)
Preparing embedding matrix.
Training model.
Train on 15998 samples, validate on 3999 samples
Epoch 1/10
15998/15998 [==============================] - 167s 10ms/step - loss: 2.4347 - acc: 0.2020 - val_loss: 1.7728 - val_acc: 0.3958
Epoch 2/10
15998/15998 [==============================] - 165s 10ms/step - loss: 1.5514 - acc: 0.4553 - val_loss: 1.2902 - val_acc: 0.5626
Epoch 3/10
15998/15998 [==============================] - 165s 10ms/step - loss: 1.1808 - acc: 0.5886 - val_loss: 1.1827 - val_acc: 0.5894
Epoch 4/10
15998/15998 [==============================] - 165s 10ms/step - loss: 0.9857 - acc: 0.6626 - val_loss: 1.0051 - val_acc: 0.6629
Epoch 5/10
15998/15998 [==============================] - 167s 10ms/step - loss: 0.8298 - acc: 0.7163 - val_loss: 1.0840 - val_acc: 0.6464
Epoch 6/10
15998/15998 [==============================] - 165s 10ms/step - loss: 0.7169 - acc: 0.7538 - val_loss: 0.9143 - val_acc: 0.7032
Epoch 7/10
15998/15998 [==============================] - 164s 10ms/step - loss: 0.6060 - acc: 0.7875 - val_loss: 0.9732 - val_acc: 0.6942
Epoch 8/10
15998/15998 [==============================] - 165s 10ms/step - loss: 0.5291 - acc: 0.8190 - val_loss: 0.8664 - val_acc: 0.7339
Epoch 9/10
15998/15998 [==============================] - 165s 10ms/step - loss: 0.4393 - acc: 0.8504 - val_loss: 0.8760 - val_acc: 0.7352
Epoch 10/10
15998/15998 [==============================] - 165s 10ms/step - loss: 0.3803 - acc: 0.8699 - val_loss: 0.8892 - val_acc: 0.7352
```

**Comment by mikkokotila** [comment]
Anything on this? It seems kind of big deal if a deep learning package is off 20% from the expected result.

**Comment by DanielDIQA** [comment]
Hi there, I second that.
I achieve on glove 300 after 100 epochs a validation accuracy of ca. 75%; training accuracy is 97%.
This is not as much as advertised... :-|
